Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1

```diff
@@ -0,0 +1,231 @@
+    }
+    else
+    {
+        Write-Warning "Could not find file $sonarReportProcessedFilePath"
+    }
+}
+
+function ProcessSonarCodeAnalysisReport
+{
+    param([string][ValidateNotNullOrEmpty()]$agentBuildDirectory)
+
+    $sonarReportFilePath = GetSonarReportFilePath $agentBuildDirectory
+    $sonarReportProcessedFilePath = GetSonarReportProcessedFilePath $agentBuildDirectory
+    
+    #read sonar-report.json file as a json object
+    $json = Get-Content -Raw $sonarReportFilePath | ConvertFrom-Json
+    Write-Verbose -Verbose "ProcessSonarCodeAnalysisReport: Total issues: $($json.issues.Count)"
+
+    ConstructComponentKeyAndPathMap $json
+
+    # '@' makes sure the result set is returned as an array
+    $newIssues = @($json.issues | Where {$_.isNew -eq $true})
+    Write-Verbose -Verbose "ProcessSonarCodeAnalysisReport: Total new issues: $($newIssues.Count)"
+
+    if ($($newIssues.Count) -gt 0)
+    {
+        foreach ($issue in $newIssues)
+        {
+            $filePath = GetRelativeFilePath $($issue.component)
+
+            #add a new property in json which stores the file path so it can be consumed directly
+            Add-Member -InputObject $issue -MemberType NoteProperty -Name relativePath -Value $filePath
+        }
+
+        #save the results into output file
+        $newIssues | ConvertTo-Json | Set-Content -Path $sonarReportProcessedFilePath
+    }
+}
+
+#creates a mapping of msbuild project guid and the path of .xxproj file on disk using ProjectInfo.xml file
+function CreateProjectGuidAndPathMap
+{
+    param([string][ValidateNotNullOrEmpty()]$agentBuildDirectory)
+
+    $parentFolder = [System.IO.Path]::Combine($agentBuildDirectory, ".sonarqube", "out")
+    $parentFolderItem = Get-Item $parentFolder
+    $directories = $parentFolderItem.GetDirectories()
+
+    Write-Host "Processing project info files..."
+
+    foreach ($directory in $directories)
+    {
+        $projectInfoFilePath = [System.IO.Path]::Combine($directory.FullName, "ProjectInfo.xml")
+        
+        if ([System.IO.File]::Exists($projectInfoFilePath))
+        {
+            Write-Verbose -Verbose "CreateProjectGuidAndPathMap: Processing project info file: $projectInfoFilePath"
+            [xml]$xmlContent = Get-Content $projectInfoFilePath
+
+            if ($xmlContent -and !$ProjectGuidAndFilePathMap.ContainsKey($xmlContent.ProjectInfo.ProjectGuid))
+            {
+                $ProjectGuidAndFilePathMap.Add($xmlContent.ProjectInfo.ProjectGuid, $xmlContent.ProjectInfo.FullPath)
+            }
+        }
+    }
+}
+
+#post-process sonar runner output (sonar-report.json) to generate code-analysis-report.json which has new issues only and repo relative file paths
+function GenerateCodeAnalysisReport
+{
+    param([string][ValidateNotNullOrEmpty()]$agentBuildDirectory)
+
+    Write-Host "Starting code analysis file path computation..."
```

Comment:
Might be confusing for the end user. Alternative "Post-processing analysis report..."